components/coins-status/index.js

```diff
@@ -1,39 +1,69 @@
-import { useEffect } from 'react';
+import { useState, useEffect } from 'react';
 import { useAnimateValue } from './useCoinStatus';
 import styles from './coin.module.css';
 import PropTypes from 'prop-types';
 
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const WALLET_URL = `${BASE_API_URL}/wallet`;
+
 const Coins = (props) => {
-  const [coins, changeCoins] = useAnimateValue(0, props.coin.value, 5000);
+  const [coins, changeCoins] = useAnimateValue(0, props.balance, 5000);
+
   useEffect(() => {
-    changeCoins(props.coin.value);
-  }, [props.coin.value, changeCoins]);
+    changeCoins(props.balance);
+  }, [props.balance, changeCoins]);
+
   return (
     <div className={styles.cointypeIndicator}>
       <div className={styles.coinData}>
-        <p> {props.coin.name}</p>
-        <p> {coins} </p>
+        <p> {props.coin}</p>
+        <p> {props.balance} </p>
```

Comment:
@rajakvk , we can destructure the props if it is repeating many times

Reply:
destructure used at suitable locations.